src/views/htmlcontent/src/js/services/data.service.ts

```diff
@@ -2,65 +2,48 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
-import {Injectable, Inject, forwardRef} from '@angular/core';
-import {Http, Headers} from '@angular/http';
-import { Observable, Subject, Observer } from 'rxjs/Rx';
-
+import { Subject } from 'rxjs/Subject';
+import { Injectable } from '@angular/core';
 import { ISlickRange } from 'angular2-slickgrid';
-
-import * as Utils from './../utils';
-
-import { ResultSetSubset, ISelectionData,
-    IResultsConfig, WebSocketEvent } from './../interfaces';
-
+import { QueryEvent, ResultSetSubset, ISelectionData } from './../interfaces';
 import * as Constants from './../constants';
-
-const WS_URL = 'ws://localhost:' + window.location.port + '/';
+import { createProxy, IMessageProtocol, IServerProxy } from '../../../../../protocol.js';
+
+declare function acquireVsCodeApi(): { postMessage: (message: string) => void; };
```

Comment:
>acquireVsCodeApi [](start = 17, length = 16)

So this gets set at run-time to the actual acquireVsCodeApi function provided by vscode?

Reply:
yes